pkg/virt-controller/watch/vmi.go

```diff
@@ -2504,3 +2527,90 @@ func statusOfReadyCondition(conditions []cdiv1.DataVolumeCondition) k8sv1.Condit
 	}
 	return k8sv1.ConditionUnknown
 }
+
+func syncIPAMClaims(vmi *virtv1.VirtualMachineInstance, kubevirtClient kubecli.KubevirtClient, hasOwnerVm bool) error {
+	if !hasOwnerVm {
+		return nil
+	}
+
+	nonAbsentIfaces := vmispec.FilterInterfacesSpec(vmi.Spec.Domain.Devices.Interfaces, func(iface virtv1.Interface) bool {
+		return iface.State != virtv1.InterfaceStateAbsent
+	})
+	nonAbsentNets := vmispec.FilterNetworksByInterfaces(vmi.Spec.Networks, nonAbsentIfaces)
+	networkNameScheme := namescheme.CreateHashedNetworkNameScheme(nonAbsentNets)
+
+	_, networkToIPAMClaimName, err := network.GetNetworkToResourceAndIPAMMap(kubevirtClient, vmi)
+	if err != nil {
+		return err
+	}
+
+	err = createIPAMClaims(vmi, nonAbsentNets, kubevirtClient, networkToIPAMClaimName, networkNameScheme)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func createIPAMClaims(vmi *virtv1.VirtualMachineInstance, nonAbsentNets []virtv1.Network, virtClient kubecli.KubevirtClient, networkToIPAMClaimName map[string]string, networkNameScheme map[string]string) error {
+	var ownerRef v1.OwnerReference
+	refs := vmi.OwnerReferences
+	for i := range refs {
+		if refs[i].Controller != nil && *refs[i].Controller {
+			ownerRef = refs[i]
+		}
+	}
+
+	for _, network := range nonAbsentNets {
+		if network.Pod != nil {
+			continue
+		} else if network.Multus != nil && network.Multus.Default {
+			continue
+		}
+		if networkToIPAMClaimName[network.Name] == "" {
+			continue
+		}
```

Comment:
IMHO this function does the wrong thing; I would give this a slice of networks for which I would have to to things, not a the whole haystack in which my function would then need to find the needles.

Please filter the list outside, and give this func exactly the inputs for which it needs to work.